exist-core/src/main/java/org/exist/xquery/functions/fn/FunXmlToJson.java

```diff
@@ -0,0 +1,246 @@
+                                    Cardinality.ZERO_OR_ONE, "The input node"),
+                            new FunctionParameterSequenceType("options", Type.MAP,
+                                    Cardinality.ONE, "The options map")
+                    },
+                    new FunctionReturnSequenceType(Type.STRING, Cardinality.ZERO_OR_ONE,
+                            "The JSON representation of the input node")
+            )
+    };
+
+    public FunXmlToJson(XQueryContext context, FunctionSignature signature) { super(context, signature); }
+
+    public Sequence eval(Sequence[] args, Sequence contextSequence) throws XPathException {
+        Sequence result;
+        final Sequence seq = (getArgumentCount() > 0) ? args[0] : Sequence.EMPTY_SEQUENCE;
+        //TODO: implement handling of options
+        final MapType options = (getArgumentCount() > 1) ? (MapType) args[1].itemAt(0) : new MapType(context);
+
+        if (seq.isEmpty())
+            {result = Sequence.EMPTY_SEQUENCE;}
+        else {
+            result = new ValueSequence();
+            final Item item = seq.itemAt(0);
+            if (item.getType() != Type.DOCUMENT && item.getType() != Type.ELEMENT) {
+                throw new XPathException(ErrorCodes.FOJS0006, "Invalid XML representation of JSON.");
+            }
+            final NodeValue nodeValue = (NodeValue) item;
+            final StringWriter stringWriter = new StringWriter();
+            nodeValueToJson(nodeValue, stringWriter);
+            final String jsonString = stringWriter.toString();
+            result.add(new StringValue(jsonString));
+        }
+        return result;
+    }
+
+    /**
+     * Generate a JSON representation of a NodeValue which conforms to
+     * https://www.w3.org/TR/xpath-functions-31/schema-for-json.xsd
+     * Traverse a NodeValue via XMLStreamReader and fill a Writer with its JSON representation
+     * by calling com.fasterxml.jackson write functions according to input type.
+     *
+     * Implements basic part of the specification. String (un)escaping is fully delegated to jackson
+     * and NOT fully conforming to spec.
+     *
+     * @param nodeValue the NodeValue to be read
+     * @param writer the Writer to be used
+     * @throws XPathException on error in XML JSON input according to specification
+     */
+    private void nodeValueToJson(NodeValue nodeValue, Writer writer) throws XPathException {
+        StringBuilder tempStringBuilder = new StringBuilder();
+        JsonFactory jsonFactory = new JsonFactory();
+        //use Stack<Object> to store String type keys and non-string type separators
+        final Integer stackSeperator = 0;
+        Stack<Object> mapkeyStack = new Stack<Object>();
+        boolean elementKeyIsEscaped = false;
+        boolean elementValueIsEscaped = false;
+        try {
+            final XMLStreamReader reader = context.getXMLStreamReader(nodeValue);
```

Comment:
should use try-with-resources to ensure it is automatically closed

Reply:
XMLStreamReader is not AutoCloseable but may throw IOException.
Do you have a suggestion?

Reply:
if its not auto-closeable, then you should use `try/finally`